web/streetmanager.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+require_once dirname(__FILE__) . '/../conf/streetmanager';
+
+class Api {
+    private $base;
+    private $token;
+
+    public function __construct($base, $username, $password, $prefix) {
+        $this->base = $base;
+        $this->token = $this->getToken($username, $password, $prefix);
+    }
+
+    private function getToken($username, $password, $prefix) {
+        $memcache = new Memcache;
+        $memcache->connect('localhost', 11211) or die ("Could not connect");
+        $token = $memcache->get($prefix . 'token');
+        if (!$token) {
+            $refresh = $memcache->get($prefix . 'refresh');
+            if ($refresh) {
+                $result = $this->call('party/refresh', 'POST', [
+                    'refresh_token' => $refresh
+                ]);
+                $token = $result->id_token;
+                $memcache->set($prefix . 'token', $token, 0, 3600);
+            } else {
+                $result = $this->call('work/authenticate', 'POST', [
+                    'emailAddress' => $username,
+                    'password' => $password
+                ]);
+                $token = $result->idToken;
+                $refresh = $result->refreshToken;
+                $memcache->set($prefix . 'token', $token, 0, 3600);
+                $memcache->set($prefix . 'refresh', $refresh, 0, 86400);
+            }
+        }
+        return $token;
+    }
+
+    public function call($url, $method, $data) {
+        $data = http_build_query($data);
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        if ($method === 'GET') {
+            curl_setopt($ch, CURLOPT_URL, $this->base . $url . '?' . $data);
+        } else {
+            curl_setopt($ch, CURLOPT_URL, $this->base . $url);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
+        }
+        if ($this->token) {
+            curl_setopt($ch, CURLOPT_HTTPHEADER, ["Token: " . $this->token]);
+        }
+        $output = curl_exec($ch);
+        curl_close($ch);
+        return json_decode($output);
+    }
+}
+
+# ---
+
+$api = new \Api(API_URL, USERNAME, PASSWORD, MEMCACHE_PREFIX);
+
+$bbox = isset($_GET['bbox']) ? explode(',', $_GET['bbox']) : [];
+if (!count($bbox)) { $bbox = [405000, 286000, 408000, 287000]; }
```

Comment:
Might be better to return an empty result if the `bbox` param is missing, saves any potential "why is this showing me roadworks in Birmingham" headaches in the future :)